File: MoveZeroes/move_zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0;
        int j = 0;
        int n = nums.length;
        while (j < n) {
            nums[i] = nums[j];
            j++;
            if (nums[i] != 0) {
                i++;
            }
        }
        while (i < n) {
            nums[i] = 0;
            i++;
        }
    }
}

File: MoveZeroes/move_zeroes.php
class Solution {
 
    function moveZeroes(&$nums) {
        $j = 0;
        for($i = 0; $i < count($nums);) {
            if ($j < count($nums)) {
                $nums[$i] = $nums[$j];
                if ($nums[$i] != 0)
                    $i++;
                $j++;
            } else {
                $nums[$i] = 0;
                $i++;
            }
        }
    }
}